src/NodeAutoCompleteViewExtension/ViewModels/NodeAutoCompleteBarViewModel.cs

```diff
@@ -970,41 +951,38 @@ internal void PopulateClusterAutoComplete()
                 Actions.Show,
                 Categories.NodeAutoCompleteOperations,
                 nameof(NodeAutocompleteSuggestion.MLRecommendation));
-
-            // Save the filtered results for search.
-            searchElementsCache = FilteredResults.ToList();
         }
 
-        internal void PopulateDefaultAutoCompleteCandidates()
+        internal IEnumerable<NodeSearchElementViewModel> DefaultAutoCompleteCandidates()
         {
             if (PortViewModel.PortModel.PortType == PortType.Input)
             {
                 switch (PortViewModel.PortModel.GetInputPortType())
                 {
                     case "int":
-                        FilteredResults = DefaultResults.Where(e => e.Name == "Number Slider" || e.Name == "Integer Slider").ToList();
-                        break;
+                        return DefaultResults.Where(e => e.Name == "Number Slider" || e.Name == "Integer Slider").ToList();
                     case "double":
-                        FilteredResults = DefaultResults.Where(e => e.Name == "Number Slider" || e.Name == "Integer Slider").ToList();
-                        break;
+                        return DefaultResults.Where(e => e.Name == "Number Slider" || e.Name == "Integer Slider").ToList();
                     case "string":
-                        FilteredResults = DefaultResults.Where(e => e.Name == "String").ToList();
-                        break;
+                        return DefaultResults.Where(e => e.Name == "String").ToList();
                     case "bool":
-                        FilteredResults = DefaultResults.Where(e => e.Name == "Boolean").ToList();
-                        break;
+                        return DefaultResults.Where(e => e.Name == "Boolean").ToList();
                     default:
-                        FilteredResults = DefaultResults.Where(e => e.Name == "String" || e.Name == "Number Slider" || e.Name == "Integer Slider" || e.Name == "Number" || e.Name == "Boolean");
-                        break;
+                        return DefaultResults.Where(e => e.Name == "String" || e.Name == "Number Slider" || e.Name == "Integer Slider" || e.Name == "Number" || e.Name == "Boolean");
                 }
             }
             else
             {
-                FilteredResults = DefaultResults.Where(e => e.Name == "Watch" || e.Name == "Watch 3D" || e.Name == "Python Script").ToList();
+                return DefaultResults.Where(e => e.Name == "Watch" || e.Name == "Watch 3D" || e.Name == "Python Script").ToList();
             }
         }
 
-        internal void OnNodeAutoCompleteWindowClosed()
+        private void SubscribeWindowEvents()
```

Comment:
With this one I was thinking we could move it to NodeAutoCompleteBarView.xaml.cs next to handling WorkspaceClosed event. There could be an argument it should be here since it's a model event but we actually only care about it in the view and we're bubbling it up from here unless I'm missing something. But I don't feel very strongly about it :)

Reply:
Ah, my idea here was that it was clearly paired with the unsubscribe. Also as you say it depends on an event in the `WorkspaceViewModel` so i opted to manage it inside the model too.